### DISCUSSION TITLE
Do we have support for NVML?

### DISCUSSION THREAD

**Discussion by saki-osive** [NVIDIA/open-gpu-kernel-modules#160] [General]
**Is your feature request related to a problem? Please describe.**
There is no support for NVML on Linux to find out the VRAM temps. 

**Describe the solution you'd like**
Does this public code provide support for that?

Please refer to [this link](https://forums.developer.nvidia.com/t/linux-operating-system-can-also-read-vram-temperature-values-like-windows/199487).

**Comment by master7720** [comment]
i am pretty sure there is support( i could be wrong tho)

**Comment by ElijahPepe** [comment]
Define support more clearly. Do you mean that the API itself is accessible via this repo, or that the underlying features in NVML (i.e. [ECC error count reporting](https://github.com/NVIDIA/open-gpu-kernel-modules/blob/main/src/common/nvswitch/kernel/inforom/ifrecc_nvswitch.c)) are in this repo? The kernel modules in this repo establish accessing the GPU, not utilizing it.

**Comment by mtijanic** [comment]
Hello, and thank you for your interest. The goal is for this driver to have full feature parity with the proprietary driver as it pertains to NVML. Currently this might not be the case due to general datacenter focus and missing features in the first release.

If you have feature requests for NVML for both drivers, the best place to make them is the [developer forum](https://forums.developer.nvidia.com/c/gpu-graphics/linux/148). Thank you.

**Comment by arisu3** [comment]
NVML does not support VRAM temperatures, only HBM temperatures on datacenter-grade GPUs.

You can use https://github.com/olealgoritme/gddr6 as a workaround, but it is not compatible with secure boot or with certain kernels that do not allow `/dev/mem` access to be restored.

Windows also does not support retrieving VRAM temperatures either from NVML or from the Windows-specific NVAPI. It's a common myth that it is supported for Windows. The truth is that some popular proprietary software for Windows implements an unsupported method of retrieving the temperatures directly from the driver using reverse engineered memory offsets, which is the same thing the above-linked repository does.
